src/test/librados_test_stub/TestClassHandler.cc

```diff
@@ -53,13 +66,19 @@ void TestClassHandler::open_all_classes() {
     ceph_abort();;
   }
 
+  std::set<std::string> names;
   struct dirent *pde = nullptr;
   while ((pde = ::readdir(dir))) {
     std::string name(pde->d_name);
+    std::cerr << name << std::endl;
     if (!boost::algorithm::starts_with(name, "libcls_") ||
         !boost::algorithm::ends_with(name, ".so")) {
       continue;
     }
+    names.insert(name);
```

Comment:
i am not sure if this makes difference.

Reply:
It pushed cls_user to the end of the init list since it seems to be related to that library being loaded first and then breaking dlopen for unrelated libraries.

Reply:
i see. but this fix is a little bit fragile: we are relying on the order of the shared objects's name, so the offending libcls_user.so is loaded after libcls_rbd.so. i think i just root caused this issue, and i am posting #21571 to fix it. could you take a look at it?

Reply:
Awesome if you found the root-cause of cls_user causing troubles (waiting for the build to finish and then I'll happily merge it). I think this is still important because it makes class loading deterministic and not random based on the directly order (which is why ARM builds sometimes work and sometimes don't).

Reply:
> which is why ARM builds sometimes work and sometimes don't

okay, this change would be a little bit paranoid, if we already have a fix.